build/yamls/elk-flow-collector/logstash/filter.rb

```diff
@@ -137,14 +137,20 @@ def filter(event)
     end
     key = event.get("[ipfix][flowKey]")
     if @@time_map.has_key?(key)
-       t = DateTime.strptime(event.get("[ipfix][flowEndSeconds]").to_s, '%Y-%m-%dT%H:%M:%S').to_time.to_i
+       t = event.get("[ipfix][flowEndSeconds]").to_i
        duration = t - @@time_map[key]
-       event.set("[ipfix][throughput]", event.get("[ipfix][octetDeltaCountFromSourceNode]").to_i / duration.to_i)
-       event.set("[ipfix][reverseThroughput]", event.get("[ipfix][reverseOctetDeltaCountFromSourceNode]").to_i / duration.to_i)
-       @@time_map[key] = t
+       # If flowEndSeconds does not change, throughput should be 0.
```

Comment:
Done. Thanks!

Reply:
Previously we were fixing the first record when start and end time are the same. This fix is discovered in scale testing where short lived connections come in as inactive flows so that multiple flows with same `flowEndSeconds` are exported.